### ISSUE TITLE
Styling not match

### ISSUE THREAD

**Issue by rainstorm-studio-australia** [resendlabs/react-email#983]
### Describe the Bug

I used Tailwind components.

It looks fine in the preview page:
![image](https://github.com/resendlabs/react-email/assets/19511189/aba2a1ba-494a-4b0b-9b05-8f53600e7ded)

But terrible on Gmail:
![image](https://github.com/resendlabs/react-email/assets/19511189/39373992-37f5-455f-868a-8518609ef393)

### Which package is affected (leave empty if unsure)

@react-email/components

### Link to the code that reproduces this issue

https://anotepad.com/notes/5nmr7a67

### To Reproduce

N/A

### Expected Behavior

Styling in Gmail, Yahoo, or any other mailing system should match the preview page

### What's your node version? (if relevant)

Node 18 LTS

**Comment by PovilasKirna** [comment]
Same I have this problem, if I render a react component it looks okay but when I send it to gmail using resend the gmail version lacks most of the styling

**Comment by calbertan** [comment]
I had this same problem when sending emails to Gmail, but Outlook emails seems to be just like the preview. This problem might be on Gmail side?

**Comment by EthanNValencia** [comment]
I have the same problem with yahoo mail.

**Comment by fewjative** [comment]
Email clients like gmail and outlook have their own styling overrides which causes your created email to look differently. I haven't found a good way to bypass that outside of using '!important' on certain properties.

**Comment by gabrielmfern** [comment]
@fewjative In what kind of properties did you have to use `!important`?

**Comment by jsonMartin** [comment]
^ I'm having the same issue, my tailwind designed email with react-email v2 looks great in the component browser and like 💩 in Gmail. It looks pretty good in Fastmail client, but still has some issues there too.

**Comment by gabrielmfern** [comment]
@luantran1311 Looking at the email  you sent, I notice a few things that cause the email to look bad here

1. The image breaks because `justify-center` and `items-center` do not have good support
2. You aren't using our Font component for fonts here so this might explain why the fonts get really broken
3. You are using the default values from Tailwind's configured utilities so it will use stuff like `rem` which doesn't have the best support

---

On another note, since most of the issues everyone is having here, that are really our responsibility, is regarding `rem` and other units for Tailwind's utilities I'll talk about that.

Although we would love to give better support and better DX on this, the best way is not to provide a hidden compilation process that would get out of your control but rather give you control with sane defaults that makes things more predictable than writing the emails in plain HTML.

That being said, we will probably make something you can add to your tailwind config optionally that addresses this and makes this a bit better to use. This also will probably come with a more guided way of defining the config for Tailwind here as currently the way we have doesn't play very well with Tailwind's VS Code extension or LSP server.

**Comment by gabrielmfern** [comment]
@jsonMartin Would love if you shared an email template maybe similar to the one you're using so that I can debug the issues you mention as well

**Comment by jsonMartin** [comment]
@gabrielmfern after looking more into it and seeing the lack of support, I think that using Tailwind `flex` attributes is what is causing my issues. Not only does it look different in Gmail, but it even looks different between Gmail in Chrome vs Firefox. 

It's a little confusing because it's so easy to make a nice email layout in the studio editor, but then what you see isn't what you get in the e-mail client. I suppose I'll just have to refactor to a Table based layout to make it work consistently between browsers.

If I still have issues after removing flex I can try to share an example template.

**Comment by jsonMartin** [comment]
I refactored my template to remove all `flex` properties. Using `table` styling gives less flexibility, but at least it works well in Gmail. There were also some other things tripping me up (negative margins not working in Gmail, etc), but I was able to find styling workarounds and it all looks good now!

As a sidenote, one enhancement that would be really helpful is if the `react-email` preview could somehow toggle stripping incompatible markup that won't work in Gmail, such as `justify-center` and negative margins. Sending an e-mail to test if it works properly in different clients is a cumbersome feedback loop.

**Comment by gabrielmfern** [comment]
> As a sidenote, one enhancement that would be really helpful is if the react-email preview could somehow toggle stripping incompatible markup that won't work in Gmail, such as justify-center and negative margins

For sure, but it's important to keep in mind the preview is just an approximation. Currently, it doesn't simulate anything about how email clients may render though.

**Comment by gabrielmfern** [comment]
> If I still have issues after removing flex I can try to share an example template.

That'd be awesome if you could

**Comment by jsonMartin** [comment]
> > As a sidenote, one enhancement that would be really helpful is if the react-email preview could somehow toggle stripping incompatible markup that won't work in Gmail, such as justify-center and negative margins
> 
> For sure, but it's important to keep in mind the preview is just an approximation. Currently, it doesn't simulate anything about how email clients may render though.

Yeah, I found that out the hard way @gabrielmfern 😆 My ignorance, but it's my first time working with `react-email` and I thought there was some magic going on to make it WYSIWYG so it would look like that in the browser too. When using Tailwind with `react-email`, it's so easy to design like you're used to; but then comes the hard part working backwards to make it render properly in Gmail. I'm not too experienced with writing safe e-mail based HTML so had to look up all the "gotchas"

But it's all good now, after removing all of the flex content and simplifying the design, I was able to get it looking almost as good. I'm just using Tailwind for stuff like `padding` and `color` now, with a mix of the `react-email` and vanilla `html` components in a table layout. 

P.S. And for sure, I know my feature suggestion is totally out of scope for this thread, just would be really convenient to have one day in a future version. I can't count how many times I spammed my Gmail inbox trying to get my last template working perfectly 😆

**Comment by aaronhawkey** [comment]
Weighing in here. Just starting to adopt the library and this should specifically be called out in the docs. My grug dev brain assumed that it was WYSIWYG sorcery.

**Comment by mrinc** [comment]
Also, the examples provided should be 100% on all different platforms (which is not the case).

**Comment by gabrielmfern** [comment]
@aaronhawkey That is a common misconception. I agree it would be a good idea. 
What would you recommend is a good place for documenting this? Would love your thoughts.

My initial thought would be answering this on some sort of FAQ, but I don't think it would be so easy to find as just opening the docs.

**Comment by gabrielmfern** [comment]
@mrinc What examples did you find did not work on what platforms?
Would love for you to open an issue on this, and I'll take a look once I can.

Would love even more if you open a PR :)

**Comment by aaronhawkey** [comment]
@gabrielmfern Thanks for the quick reply! You rock! 

Truthfully I am not sure since I'm the type of dev that rushes to try to implement, then references the docs when he gets stuck. But it might help to have some sort of compatibility guide on the Tailwind component docs. What would work and what wouldn't with certain clients? That's prob a lot to ask. I would also maybe move the disclaimer higher up towards the top about incompatibility. 

A lot of people probably won't have a strong foundation on writing html emails and the underlying complexity (like me). How do you best communicate that? I'm not certain.

**Comment by jsonMartin** [comment]
Just to chime in to the conversation about WSYWIG misconception @gabrielmfern @aaronhawkey — I think it's much easier to fall into this trap when using Tailwind, because web developers will likely bring their existing mental models with them, write the Tailwind code they're used to writing, and expect it to work. And with the responsive e-mail studio editor with visual feedback confirming that feeling, it makes it easy to keep working on a great looking template until realizing a few hours later it doesn't actually work in Outlook 😆 

I think putting a callout somewhere prominent in the Tailwind documentation (that it compiles down to normal HTML and should only be used for classes with support for e-mail clients) could save developers like myself a lot of time going down the wrong path. FAQ would be good too, maybe if just to mention that there is no WSYWIG magic, make sure to use proper HTML email and provide a link to a site like https://www.caniemail.com/ for supported elements

**Comment by eaoliver** [comment]
This is a useful resource for anyone encountering this issue: [https://developers.google.com/gmail/design/css](https://developers.google.com/gmail/design/css )

**Comment by bhatvikrant** [comment]
I've realised it's best to use the style tag for layout-ing and TailwindCSS for stuff like padding, margin and colors for now.

**Comment by paschalidi** [comment]
Hm, this is the issue i faced today as well.

I am wondering if there is any workaround on this? I had to re-write all my emails with tables, which really defeats the purpose of using tailwind for styling.

Would be very keen to hear  if anyone came up with a smart workaround?

**Comment by ktlyee** [comment]
I have been faced with this issue when I sent an email to Yahoo. It seems like CSS padding is not working as expected. But Gmail and Outlook work fine. I'm not sure that Yahoo isn't support or has limitations for rendering CSS. 

Does anyone have a solution to fix this issue?

**Comment by gabrielmfern** [comment]
@ktlyee I don't think Yahoo supports padding actually, from my experience it always didn't respect any padding I tried setting. What I try to do generally is use a margin instead if possible, or just aligning things instead.

**Comment by matiasdecarli** [comment]
I had this same issue. Solved it by centering the image like this

```
<Img
	src="https://..."
	alt="description"
	width={75}
	height={45}
	style={{ display: 'block', margin: '0 auto' }}
/>
```

**Comment by gabrielmfern** [comment]
Closing this since we integrated a pixel-based preset for Tailwind and updated our Tailwind examples to use it. It should fix the `rem` issues.
